tests/nc-client/main.c

```diff
@@ -0,0 +1,175 @@
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#include "events.h"
+#include "network.h"
+#include "sock.h"
+#include "warnp.h"
+
+struct senddata {
+	char * buffer;
+	ssize_t nchars;
+	int socket;
+	int conndone;
+	void * connect_cookie;
+	void * write_cookie;
+};
+
+/* Forward declarations. */
+static int send_input(void *);
```

Comment:
Not necessary, since the function is defined before it's first used.  I think it's only `callback_wrote` which needs a forward declaration.